sklearn/preprocessing/tests/test_data.py

```diff
@@ -968,6 +968,37 @@ def test_quantile_transform_check_error():
                          transformer.transform, 10)
 
 
+@pytest.mark.parametrize(
+    "missing_values, dtype",
+    [(np.nan, np.float64)])
```

Comment:
Please also test for `(np.nan, np.float32)` and check that the output has `dtype == np.float32`.